server/http.go

```diff
@@ -126,6 +126,12 @@ func (p *Plugin) completeUserOAuthToZoom(w http.ResponseWriter, r *http.Request)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+	} else {
+		err = p.setUserToken(userID, token)
```

Comment:
This seems cleaner so far, but a few lines after (line 161) we are storing the token along the OAuthUserInfo. If we want to follow this pattern, we would need to refactor the token out of the OAuthUserInfo and fix whatever breaks from that.

Another solution would be to keep the current pattern, and just update the token when needed in the OAuthUserInfo structure when needed.

I leave up to you which path to take.